frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_detail/widgets/mobile_row_property_list.dart

```diff
@@ -0,0 +1,220 @@
+
+            context.read<RowDetailBloc>().add(
+                  RowDetailEvent.reorderField(
+                    reorderedFieldId,
+                    targetFieldId,
+                    oldIndex,
+                    newIndex,
+                  ),
+                );
+          },
+          footer: Padding(
+            padding: const EdgeInsets.symmetric(vertical: 16),
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [
+                if (context.read<RowDetailBloc>().state.numHiddenFields != 0)
+                  const ToggleHiddenFieldsVisibilityButton(),
+                const VSpace(8),
+                // add new field
+                MobileCreateRowFieldButton(
+                  viewId: viewId,
+                ),
+              ],
+            ),
+          ),
+          children: children,
+        );
+      },
+    );
+  }
+}
+
+class _PropertyCell extends StatefulWidget {
+  final DatabaseCellContext cellContext;
+  final GridCellBuilder cellBuilder;
+
+  final int index;
+  const _PropertyCell({
+    required this.cellContext,
+    required this.cellBuilder,
+    Key? key,
+    required this.index,
+  }) : super(key: key);
+
+  @override
+  State<StatefulWidget> createState() => _PropertyCellState();
+}
+
+class _PropertyCellState extends State<_PropertyCell> {
+  @override
+  Widget build(BuildContext context) {
+    const cellHeight = 32.0;
+    final style = _customCellStyle(widget.cellContext.fieldType);
+    final cell = widget.cellBuilder.build(widget.cellContext, style: style);
+
+    return ListTile(
+      contentPadding: EdgeInsets.zero,
+      visualDensity: VisualDensity.compact,
+      horizontalTitleGap: 4,
+      // FieldCellButton in Desktop
+      // TODO(yijing): adjust width with sreen size
+      leading: SizedBox(
+        width: 150,
+        height: cellHeight,
+        child: TextButton.icon(
+          icon: FlowySvg(
+            widget.cellContext.fieldInfo.field.fieldType.icon(),
+            color: Theme.of(context).colorScheme.onSurface,
+          ),
+          label: Text(
+            widget.cellContext.fieldInfo.field.name,
+            // TODO(yijing): update text style
+            style: Theme.of(context).textTheme.bodyMedium?.copyWith(
+                  color: Theme.of(context).colorScheme.onSurface,
+                ),
+            overflow: TextOverflow.ellipsis,
+          ),
+          style: TextButton.styleFrom(
+            alignment: Alignment.centerLeft,
+            padding: EdgeInsets.zero,
+          ),
+          // naivgator to field editor
+          onPressed: () => context.push(
+            CardPropertyEditScreen.routeName,
+            extra: {
+              CardPropertyEditScreen.argCellContext: widget.cellContext,
+              CardPropertyEditScreen.argRowDetailBloc:
+                  context.read<RowDetailBloc>(),
+            },
```

Comment:
I don't think it's a good way to pass the block and context through the router. The parameters should be basic types, like string and int, and then you can create the necessary block and other components.

Reply:
I agree. I plan to refactor here while working on the board view. Currently, this part code still depends on the shared code with the desktop. I probably will refract the upper layers of code to fit the mobile needs.